ProcessOptimizer/samplers/doe/optimal_design.py

```diff
@@ -0,0 +1,697 @@
+        X, design, factor_names, code, res=res, space=space
+    )
+
+    return opt_design
+
+
+def model_order_and_include_powers(design_type):
+    """
+    Determine the order and include_powers of the model based on the design
+    type
+
+    :param design_type: The design_type of design to create.
+    :type design_type: str
+    :options: 'linear', 'screening', 'response', 'optimization'
+
+    :return: The order and include_powers of the model
+    :rtype: int, bool
+    """
+
+    # Specify options for the design types
+    design_types = ["linear", "screening", "response", "optimization"]
+    model_orders = [1, 2, 2, 3]
+    model_include_powers = [None, False, True, True]
+
+    if design_type is not None and design_type not in design_types:
+        raise ValueError(f"design_type must be one of {design_types}")
+
+    # Determine the order and "include powers" of the model
+    if design_type is not None:
+        design_model_orders = dict(zip(design_types, model_orders))
+        order = design_model_orders[design_type]
+        include_powers_dict = dict(zip(design_types, model_include_powers))
+        include_powers = include_powers_dict[design_type]
+    else:
+        order = None
+        include_powers = None
+
+    return order, include_powers
+
+
+def get_cat_var_levels(factor_space):
+    """
+    Determine the levels of the categorical variables in the factor space.
+    Gives a list of the number of levels for each factor in the factor space.
+    List element will be None if the factor is not categorical
+
+    :param factor_space: The space of the factors
+    :type factor_space: ProcessOptimizer.space.Space
+
+    :return: The levels of the categorical variables in the factor space
+    :rtype: list of int or None
+    """
+
+    var_is_cat = []
+
+    for factor in factor_space.dimensions:
+        if isinstance(factor, Categorical):
+            # categorical_options *= len(factor.categories)
+            # INITIALLY, only allow 2 level categorical factors
+            var_is_cat.append(len(factor.categories))
+            if len(factor.categories) != 2:
+                raise ValueError(
+                    "Only 2 level categorical factors are supported"
+                )
+        else:
+            var_is_cat.append(None)
+    return var_is_cat
+
+
+def include_powers_to_list(include_powers, cat_var_levels):
+    """
+    Convert the include_powers parameter to a list if True and there are
+    categorical variables
+
+    :param include_powers: Whether to include squared and cubed terms
+    :type include_powers: bool, list of bool
+
+    :param cat_var_levels: The levels of the categorical variables in the
+        factor space
+    :type cat_var_levels: list of int or None
+
+    :return: The include_powers parameter as a list
+    :rtype: list of bool
+    """
+
+    if include_powers is True and any(level for level in cat_var_levels):
+        include_powers = [True] * len(cat_var_levels)
```

Comment:
```suggestion
        include_powers = [True if level is None else False for level in cat_var_levels]
        # cat_var_levels[i] is None for non-categorical levels. We only include powers for non-categorcial levels
```
This is partly me guessing on what the purpose of this function is, let me know if I got it right.

Reply:
Potentially a much cleaner solution but not sure if it breaks some edge cases. e.g. having categorical and not including power terms as False or None